src/conduits/RwaCounduits3.integration.t.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-FileCopyrightText: © 2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "forge-std/Test.sol";
+import "ds-token/token.sol";
+import "ds-math/math.sol";
+import "ds-value/value.sol";
+import "dss-interfaces/Interfaces.sol";
+
+import {DssPsm} from "dss-psm/psm.sol";
+
+import {RwaInputConduit3} from "./RwaInputConduit3.sol";
+import {RwaOutputConduit3} from "../conduits/RwaOutputConduit3.sol";
+
+abstract contract RwaConduits3TestAbstract is Test, DSMath {
+    // Define both in constructor of derived contract
+    bytes32 ILK;
+    address psm;
+
+    address me = address(this);
+    address testUrn = vm.addr(420);
+
+    ChainlogAbstract changelog = ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+    VatAbstract vat = VatAbstract(changelog.getAddress("MCD_VAT"));
+
+    DaiAbstract dai;
+    GemAbstract gem;
+
+    uint256 GEM_DECIMALS;
+    uint256 GEM_DAI_DIFF_DECIMALS;
+    uint256 ART_LEFT;
```

Comment:
Since these are not constants, they should be `camelCased`.

Reply:
It's kind of a constant, the thing is that I can't make it a real constant because I have of dynamic part there. Can I explain more if you want, but I prefer to leave it like it is.

Reply:
I see.